### PR TITLE
Cleanup client not found errors

### PR DESCRIPTION
**Pull request by dnephin**
Fixes #34782

Add NotFound errors to all Remove calls
Remove redundant notFound error implementations.

### CODE REVIEW

**Comment by dnephin** [comment]
Oh , I guess I can't change the errors expected in the cli tests because those wont change until the cli gets this update.  I will have to revert that part.

**Review by boaz0 — APPROVED** [review]
LGTM

File: client/errors.go [hunk 1]
```diff
@@ -144,70 +110,22 @@ func IsErrUnauthorized(err error) bool {
...
-
-// Error returns a string representation of a serviceNotFoundError
-func (e serviceNotFoundError) Error() string {
-	return fmt.Sprintf("Error: No such service: %s", e.serviceID)
-}
-
-// NotFound indicates that this error type is of NotFound
-func (e serviceNotFoundError) NotFound() bool {
-	return true
+	return IsErrNotFound(err)
```
Comment (aaronlehmann) [inline client/errors.go]: Are these functions useful if they all return `IsErrNotFound(err)`? It seems a bit misleading to imply they check for a specific type of not-found error.

File: integration-cli/docker_api_images_test.go [hunk 2]
```diff
@@ -93,7 +93,7 @@ func (s *DockerSuite) TestAPIImagesDelete(c *check.C) {
 	c.Assert(err.Error(), checker.Contains, "unable to delete")
 
 	_, err = cli.ImageRemove(context.Background(), "test:noexist", types.ImageRemoveOptions{})
-	c.Assert(err.Error(), checker.Contains, "No such image")
+	c.Assert(err.Error(), checker.Contains, "no such image")
```
Comment (simonferquel) [inline integration-cli/docker_api_images_test.go]: Might be usefull to check with client.IsNotFound(err) here (and do that for every api/remove tests)

File: integration-cli/docker_api_stats_test.go [hunk 3]
```diff
@@ -266,7 +266,7 @@ func (s *DockerSuite) TestAPIStatsContainerNotFound(c *check.C) {
 	c.Assert(err, checker.IsNil)
 	defer cli.Close()
 
-	expected := "No such container: nonexistent"
+	expected := "no such container: nonexistent"
```
Comment (simonferquel) [inline integration-cli/docker_api_stats_test.go]: Same as for images, check that client.IsNotFound returns true

**Review by simonferquel — CHANGES_REQUESTED** [review]
Add / modify API tests doing get/update/remove on every resource type to make sure client.IsErrNotFound(err) returns true

File: client/errors.go (hunk 1 above)
Comment (dnephin) [reply client/errors.go]: They are not useful, which is why they were deprecated in #34767. Unfortunately they are commonly used, so we can't just removed them immediately.

I'll rebase this to pickup the changes in #34767.

File: integration-cli/docker_api_images_test.go (hunk 2 above)
Comment (dnephin) [reply integration-cli/docker_api_images_test.go]: This is already done in the `client/` unit tests, which I think it the appropriate place to do this. The API tests should be testing the API not the client.

**Comment by dnephin** [comment]
rebased

**Comment by dnephin** [comment]
windows CI failure is unrelated and fixed in master

**Review by vdemeester — APPROVED** [review]
LGTM 🐮

**Comment by thaJeztah** [comment]
@dnephin is this fixing something that we need to backport to a Docker release as well?

**Comment by dnephin** [comment]
Not really, mostly just a fix for windows CI.

**Comment by thaJeztah** [comment]
Perfect, thanks!

**Pull request merged**